avm/utilities/pipelines/platform/Invoke-AvmJsonModuleIndexGeneration.ps1

```diff
@@ -0,0 +1,169 @@
+    }
+
+    ## Check if the last version of the moduleIndex.json (last-moduleIndex.json) file exists and is not empty
+
+    if (Test-Path $lastModuleIndexJsonFilePath) {
+      $lastModuleIndexJsonFileContent = Get-Content $lastModuleIndexJsonFilePath
+      if ($null -eq $lastModuleIndexJsonFileContent) {
+        Write-Error "The last version of the moduleIndex.json file (last-moduleIndex.json) exists but is empty. File: $lastModuleIndexJsonFilePath" -ErrorAction Stop
+      }
+      Write-Verbose 'The last version of the moduleIndex.json file (last-moduleIndex.json) exists and is not empty. Proceeding...' -Verbose
+    }
+  }
+
+  ## Generate the new moduleIndex.json file based off the modules in the repository
+
+  $currentGeneratedModuleIndexJsonFilePath = $prefixForCurrentGeneratedModuleIndexJsonFile + $moduleIndexJsonFilePath
+
+  Write-Verbose "Generating the current generated moduleIndex.json file and saving to: $currentGeneratedModuleIndexJsonFilePath ..." -Verbose
+
+  $moduleIndexData = @()
+
+  foreach ($avmModuleRoot in @('avm/res', 'avm/ptn')) {
+    $avmModuleGroups = (Get-ChildItem -Path $avmModuleRoot -Directory).Name
+
+    foreach ($moduleGroup in $avmModuleGroups) {
+      $moduleGroupPath = "$avmModuleRoot/$moduleGroup"
+      $moduleNames = (Get-ChildItem -Path $moduleGroupPath -Directory).Name
+
+      foreach ($moduleName in $moduleNames) {
+        $modulePath = "$moduleGroupPath/$moduleName"
+        $mainJsonPath = "$modulePath/main.json"
+        $tagListUrl = "https://mcr.microsoft.com/v2/bicep/$modulePath/tags/list"
+
+        try {
+          Write-Verbose "Processing AVM Module '$modulePath'..." -Verbose
+          Write-Verbose "  Getting available tags at '$tagListUrl'..." -Verbose
+
+          try {
+            $tagListResponse = Invoke-RestMethod -Uri $tagListUrl
+          } catch {
+            Write-Output "Error occurred while accessing URL: $tagListUrl"
+            Write-Output "Error message: $($_.Exception.Message)"
+            continue
+          }
+          $tags = $tagListResponse.tags | Sort-Object
+
+          $properties = [ordered]@{}
+          foreach ($tag in $tags) {
+            $gitTag = "$modulePath/$tag"
+            $documentationUri = "https://github.com/Azure/bicep-registry-modules/tree/$gitTag/$modulePath/README.md"
+
+            try {
+              $moduleMainJsonUri = "https://raw.githubusercontent.com/Azure/bicep-registry-modules/$gitTag/$mainJsonPath"
+              Write-Verbose "    Getting available description for tag $tag via '$moduleMainJsonUri'..." -Verbose
+              $moduleMainJsonUriResponse = Invoke-RestMethod -Uri $moduleMainJsonUri
+              $description = $moduleMainJsonUriResponse.metadata.description
+            } catch {
+              Write-Output "Error occurred while accessing description for tag $tag via '$moduleMainJsonUri'"
+              Write-Output "Error message: $($_.Exception.Message)"
```

Comment:
Maybe we should use a `write-error` (unless that pushes us right into the next try catch). But ideally, as long as the script is not invoked with an `'-ErrorAction 'Stop'` this should work.

But that aside, I'd suggest to set an error variable that at the beginning of the script is `$false`, is set to `$true`  in any block that invokes a `continue` and returns it back to the pipeline. There, we should publish it into the pipeline and throw an exception after the publishing to the storage account in the subsequent task. Reason being: Only if we do that we'd actually know something is off - without blocking the moduleIndex publishing.

Thoughts?

Reply:
I initially belived that write-error in try-catch invoked a terminating error. But upon re-investigation it does not, so have updated.

Looking at the second part now as i agree its a good idea to log a failing pipeline